tests/Test/Xrefcheck/TrailingSlashSpec.hs

```diff
@@ -0,0 +1,45 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.TrailingSlashSpec where
+
+import qualified Unsafe
+
+import Test.Hspec (Spec, describe, it, runIO)
+import Test.QuickCheck ((===))
+
+import Xrefcheck.Config
+import Xrefcheck.Core
+import Xrefcheck.Progress
+import Xrefcheck.Scan
+import Xrefcheck.Scanners.Markdown
+import Xrefcheck.Verify
+
+spec :: Spec
+spec = do
+  describe "Trailing forward slash detection" $ do
+    let config = defConfig GitHub
+    let format = specificFormatsSupport [markdownSupport (scMarkdown (cScanners config))]
+    forM_ roots $ \root -> do
+      repoInfo <- runIO $ allowRewrite False $ \rw ->
+        gatherRepoInfo rw format TraversalConfig{ tcIgnored = [] } root
+      res <- runIO $ allowRewrite False $ \rw ->
+        verifyRepo rw (cVerification config) FullMode root repoInfo
+      it ("Analysis results for the root with" <>
+          (if Unsafe.last root == '/' then "" else "out") <>
+          " the trailing forward slash should not contain the FileDoesNotExist error") $ do
+        let satisfactoryCase = maybe True (not . any checkFileDoesNotExist) $ verifyErrors res
+        satisfactoryCase === True
+  where
+    checkFileDoesNotExist :: WithReferenceLoc VerifyError -> Bool
+    checkFileDoesNotExist WithReferenceLoc{..} = case wrlItem of
+      FileDoesNotExist _ -> True
+      _ -> False
+
+    roots :: [String]
+    roots =
+      [ "tests/markdowns/without-annotations"
+      , "tests/markdowns/without-annotations/"
+      ]
```

Comment:
These tests seem to assume that all files in the `tests/markdowns/without-annotation` folder are valid, and thus `verifyRepo` should pass.

That's a valid assumption _at the moment_, but the point of these folders is to contain some sample markdown files we want to test which may or may not be valid.

A solution might be to _not_ call `verifyRepo`. We could simply check that all the filepaths in `repoInfo` are indeed valid, i.e., they exists on disk.

Reply:
I see @Martoon-00 already made a similar point above :smile:

Reply:
I stand corrected, it seems the tests make no such assumption. But still, it would be simpler to just check that the files from `repoInfo` exist on disk.